### PR TITLE
fix(ng-select-comp): handle IME compositon mode

### PR DESCRIPTION
**Pull request by ZeekoZhu**
Stop filter search term when IME is in compostion mode, and filter search term on IME quit composition mode.

Fix: #704

*Chinese IME*
Chinese

*Japanese IME*
Japanese

### CODE REVIEW

File: src/ng-select/lib/ng-select.component.ts [hunk 1]
```diff
@@ -515,7 +522,17 @@ export class NgSelectComponent implements OnDestroy, OnChanges, AfterViewInit, C
         return empty && this._isTypeahead && !this.searchTerm && !this.loading;
     }
 
+    handleSearchComposition(isCompositionStart: boolean, term: string) {
```
Comment (varnastadeus) [inline src/ng-select/lib/ng-select.component.ts]: split this method to separate methods, eg:. `onCompositionStart` and `onCompositionEnd`

File: src/ng-select/lib/ng-select.component.ts [hunk 2]
```diff
@@ -191,6 +191,13 @@ export class NgSelectComponent implements OnDestroy, OnChanges, AfterViewInit, C
     private _onChange = (_: any) => { };
     private _onTouched = () => { };
 
+    /**
+     * Is IME in compostion mode
+     * 
+     * @memberof NgSelectComponent
+     */
+    private _IMEComposition = false;
```
Comment (varnastadeus) [inline src/ng-select/lib/ng-select.component.ts]: `IME` abbreviation is not very common so it adds more confusion to variable name than information, I would suggest naming it simpler like `_isComposing` or `_composing`. Also, remove JSDoc comments as other members don't have them.

File: src/ng-select/lib/ng-select.component.ts [hunk 3]
```diff
@@ -515,7 +522,17 @@ export class NgSelectComponent implements OnDestroy, OnChanges, AfterViewInit, C
...
+    handleSearchComposition(isCompositionStart: boolean, term: string) {
+        this._IMEComposition = isCompositionStart;
+        if (this._IMEComposition === false) {
+            this.filter(term);
+        }
+    }
+
     filter(term: string) {
+        if (this._IMEComposition) {
+            return;
```
Comment (varnastadeus) [inline src/ng-select/lib/ng-select.component.ts]: this condition probably needs to be added to other cases also - when the user presses `enter` or `tab` it should skip until the composition is ended.

Comment (ZeekoZhu) [reply src/ng-select/lib/ng-select.component.ts]: I did some tests, browser will not receive `tab` or `enter` during compotion.

File: src/ng-select/lib/ng-select.component.ts (hunk 2 above)
Comment (ZeekoZhu) [reply src/ng-select/lib/ng-select.component.ts]: Fixed in ab098331563ac33f6e50204f9088a491c7742493

File: src/ng-select/lib/ng-select.component.ts (hunk 1 above)
Comment (ZeekoZhu) [reply src/ng-select/lib/ng-select.component.ts]: Fixed in ab098331563ac33f6e50204f9088a491c7742493

**Pull request merged**